arch/lkl/kernel/syscalls.c

```diff
@@ -342,24 +336,17 @@ int initial_syscall_thread(void *sem)
 {
 	int ret = 0;
 
-	if (lkl_ops->tls_alloc)
-		ret = lkl_ops->tls_alloc(&syscall_thread_data_key);
-	if (ret)
-		return ret;
-
 	init_pid_ns.child_reaper = 0;
 
 	ret = syscall_thread_data_init(&default_syscall_thread_data, sem);
 	if (ret)
 		goto out;
 
+	syscall_thread_data = &default_syscall_thread_data;
```

Comment:
I think this is unnecessary. default syscall thread is a special kernel thread that is not attached to any host thread. Also here, it's a kernel thead instead of host thread running this function.